crates/rbuilder/src/bin/debug-fake-order-sim.rs

```diff
@@ -39,18 +42,27 @@ pub async fn main() -> eyre::Result<()> {
         cancel.clone(),
     );
 
-    tokio::spawn(async move {
-        while let Some(command) = sim_results.orders.recv().await {
-            println!("{:?}", command);
-        }
-    });
-
-    let tx_args = TxArgs::new_send_to_coinbase(NamedAddr::User(1), 0, 5);
+    // Create a simple tx that sends to coinbase 5 wei.
+    let coinbase_profit = 5;
+    // max_priority_fee will be 0
+    let tx_args = TxArgs::new_send_to_coinbase(NamedAddr::User(1), 0, coinbase_profit);
     let tx = test_context.sign_tx(tx_args)?;
     let tx = TransactionSignedEcRecoveredWithBlobs::new_no_blobs(tx).unwrap();
     order_sender.send(OrderPoolCommand::Insert(Order::Tx(MempoolTx::new(tx))))?;
-
-    tokio::time::sleep(Duration::from_secs(5)).await;
+    // We expect to receive the simulation giving a profit of coinbase_profit since that's what we sent directly to coinbase.
+    // and we are not paying any priority fee
+    if let Some(command) = sim_results.orders.recv().await {
+        match command {
+            SimulatedOrderCommand::Simulation(sim_order) => {
+                println!("Got the sim order!! SimValue = {:?}", sim_order.sim_value);
+                assert_eq!(
+                    sim_order.sim_value.coinbase_profit,
+                    U256::from(coinbase_profit)
+                );
```

Comment:
maybe add an error message in case this panics?
assert_eq!(
                    sim_order.sim_value.coinbase_profit,
                    U256::from(coinbase_profit),
"Simulated profit {} differs from value sent to coinbase {}",                     sim_order.sim_value.coinbase_profit,
                    U256::from(coinbase_profit),
                );